public/js/index.js

```diff
@@ -524,6 +524,62 @@ Index.loadContextMenuCategories = (catList, id) => Server.callAPI(`/api/archives
     },
 );
 
+/**
+ * Build rating options for contextMenu and select the one for the current ID.
+ * @param {*} id The ID of the archive to check
+ * @returns Ratings
+ */
+Index.loadContextMenuRatings = (id) => Server.callAPI(`/api/archives/${id}/metadata`, "GET", null, `Error finding metadata for ${id}!`,
+    (data) => {
+        const items = {};
+        const ratings = [{
+            id: 1,
+            name: "⭐",
+        }, {
+            id: 2,
+            name: "⭐⭐",
+        }, {
+            id: 3,
+            name: "⭐⭐⭐",
+        }, {
+            id: 4,
+            name: "⭐⭐⭐⭐",
+        }, {
+            id: 5,
+            name: "⭐⭐⭐⭐⭐",
+        }];
+        const tags = data.tags.split(",").map((x) => x.split(":"));
+        const hasRating = tags.some((x) => x[0] === "rating");
+
+        items.push({
+            name: "",
+            events: {
+                click() {
+                    Server.updateTagsFromArchive(id, tags.filter((x) => x[0] !== "rating").map((x) => x.join(":")).join(","));
+                },
+            },
+        });
```

Comment:
This block fails for me on Firefox as `push` only works with arrays, not objects.  
Also, I'm not quite sure what this click event even does?

Reply:
Fixed on new commit.